src/main/java/io/citrine/jcc/search/analysis/result/HistogramAnalysisResult.java

```diff
@@ -0,0 +1,111 @@
+package io.citrine.jcc.search.analysis.result;
+
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+
+/**
+ * Results of a histogram analysis.
+ *
+ * @author Kyle Michel
+ */
+public class HistogramAnalysisResult extends AnalysisResultWithBuckets<HistogramAnalysisResult.Bucket> {
+
+    @Override
+    public HistogramAnalysisResult setBuckets(final List<HistogramAnalysisResult.Bucket> buckets) {
+        super.setBuckets(buckets);
+        return this;
+    }
+
+    @Override
+    public HistogramAnalysisResult addBuckets(final List<HistogramAnalysisResult.Bucket> buckets) {
+        super.addBuckets(buckets);
+        return this;
+    }
+
+    @Override
+    public HistogramAnalysisResult addBuckets(final HistogramAnalysisResult.Bucket buckets) {
+        super.addBuckets(buckets);
+        return this;
+    }
+
+    @Override
+    public boolean equals(final Object rhs) {
+        if (this == rhs) {
+            return true;
+        }
+        if ((rhs == null) || !(rhs instanceof HistogramAnalysisResult)) {
+            return false;
+        }
+        final HistogramAnalysisResult rhsResult = (HistogramAnalysisResult) rhs;
+        return super.equals(rhsResult);
+    }
+
+    /**
+     * Bucket to store information from a histogram.
+     *
+     * @author Kyle Michel
+     */
+    public static class Bucket extends AnalysisResultWithBuckets.Bucket {
```

Comment:
I know its not actually part of the ES response, but I think having the interval or max value stored in the bucket would be very helpful in many cases.

Reply:
That could also let you unify the response object with the range response.

Reply:
Sean and I agree but it is actually a lot of work to do this. Let's keep it in mind as an improvement for later.